packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -128,6 +89,15 @@ const toggleBlock = (editor, value) => {
   };
 
   Transforms.setNodes(editor, newProperties);
+
+  const currentSelection = editor.selection;
+  Transforms.deselect(editor);
+
+  // setTimeout is required to highlight and focus to take effect, https://github.com/ianstormtaylor/slate/issues/3412
+  setTimeout(() => {
+    Transforms.select(editor, currentSelection);
+    ReactEditor.focus(editor);
+  }, TIMEOUT);
```

Comment:
Where did you get the value of timeout from? 🤔 

Isn't this the same issue as I fixed in #18125 

CC @markkaylor & @Feranchz

Reply:
I do not believe Editor vs ReactEditor is the issue, they are different things that serve different purposes. The slate examples use both of them the same way we are. 

I have tested Slate APIs like ReactEditor.focus(), ReactEditor.toDOMNode, Transforms.insertNode, etc... in an isolated Slate environment and they are working without issue. I believe the issues we are facing are related to our environment and are not caused by Slate. 

The issues with focusing seems to be around fighting the design system for focus. For example here we have the select dropdown we click an option and then call ReactEditor.focus(). The editor in fact gets the focus, but then loses it since the SelectTrigger then calls focus. (not sure how that order of operations is determined 🤔)
 
For instance, removing these two lines:
https://github.com/strapi/design-system/blob/ccff1ce3da33c234ae21ad59e79675f45c3b48c1/packages/primitives/src/components/Select/Select.tsx#L814-L819

Solves the issue when calling ReactEditor.focus() here with no need for a timeout.

Obviously we can't do that and currently I'm not sure how we ensure the ReactEditor.focus() takes precedence and the focus from the design system is ignored. Maybe you have some ideas @joshuaellis 

For the other issue on Fernando's PR with ReactEditor.toDOMNode, I believe React.useEffect is still a workaround but more appropriate than setTimeout in my opinion. 

We are still investigating but I am personally feeling more confident that the issue is not related to Slate, as mentioned above ReactEditor.toDOMNode works fine in an isolated Slate environment. See this sandbox for example: https://codesandbox.io/s/determined-moon-6kxd56?file=/src/App.js

All that being said I would be happy to pair with you @joshuaellis whenever you are available to come up with a plan for a sustainable solution.

Reply:
Great investigating – feels like we might have found the issues 🤓 lets setup a meet for tomorrow AM and we can sort this out and get moving again? 🙏🏼